core/model/src/main/java/org/eclipse/sw360/antenna/model/util/ArtifactUtils.java

```diff
@@ -76,4 +78,23 @@ private ArtifactUtils() {
                         .map(fn -> new GenericArtifactCoordinates(fn, "-")),
                 artifact);
     }
+
+    public static ArtifactCoordinates createArtifactCoordinates(PackageURL packageURL) {
+        String type = packageURL.getType();
+
+        switch (type) {
+            case "maven":
+                return new MavenCoordinates(packageURL.getName(), packageURL.getNamespace(), packageURL.getVersion());
+            case "npm":
+                return new JavaScriptCoordinates(packageURL.getName(), packageURL.getNamespace(), packageURL.getVersion());
+            case "nuget":
+                return new DotNetCoordinates(packageURL.getName(), packageURL.getVersion());
+            default:
+                return new GenericArtifactCoordinates(packageURL.getName(), packageURL.getVersion());
+        }
+    }
+
+    public static ArtifactCoordinates createArtifactCoordinates(String packageUrl) throws MalformedPackageURLException {
```

Comment:
Please do not throw the very specific exception. My proposal would be to use a IllegalArgumentException, since the String is not fulfilling the precondition of the packageUrl parameter.